go/vt/vtgate/planbuilder/symtab.go

```diff
@@ -109,15 +111,38 @@ func (st *symtab) AddVindexTable(alias sqlparser.TableName, vindexTable *vindexe
 		origin:  rb,
 	}
 
-	// Pre-create vindex columns
+	for _, col := range vindexTable.Columns {
+		t.columns[col.Name.Lowered()] = &column{
+			origin: rb,
+			name:   col.Name,
+			typ:    col.Type,
+			table:  t,
+		}
+	}
+
 	for _, cv := range vindexTable.ColumnVindexes {
-		c := &column{
+		lowered := cv.Column.Lowered()
+		if c, ok := t.columns[lowered]; ok {
```

Comment:
Should it be named `col` instead of `c`, so it's consistent with naming above?

Reply:
I've rebased this branch with your multi-column changes, which required me to rewrite this logic. I've also renamed the var.